### DISCUSSION TITLE
Not easy to include in bazel projects

### DISCUSSION THREAD

**Discussion by sidewaiise** [eProsima/Fast-DDS#3202] [Ideas]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

It shoud be easy to include this as a package in a Bazel project. Currently bazel is not supported out of the box.

Enabling this would allow c/c++ & other projects leverage fastdds easily via a bazel dependency

### Current behavior

No Bazel config

### Steps to reproduce

Try including this in a single bazel dependency block - wont' work

### Fast DDS version/commit

HEAD

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by JLBuenoLopez** [comment]
Hi @sidewaiise 

According to [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#feature-request), improvements suggestions should be opened in the [Ideas discussion forum](https://github.com/eProsima/Fast-DDS/discussions/categories/ideas). I am going to proceed and move this issue there.

**Comment by EduPonz** [comment]
Hi @sidewaiise ,

We do not posses much knowledge about Bazel right now, nor is it on our roadmap to support it. That being said, we are open to contribution in this or any other regard, specially if you have the knowledge to do so.

**Reply by EduPonz** [reply]
I just saw #3200!

**Reply by sidewaiise** [reply]
Welcome to any feedback here. I think most of the controversy will be around supported platforms, architectures and dependency versions.

I'll also add a README item on use with Bazel. This might cut the learning curve and also inform maintainers here :)

**Reply by JLBuenoLopez** [reply]
Hi @sidewaiise,

Fast DDS project defines the [supported platforms and architectures](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md). Regarding the dependencies, each Fast DDS release includes the [`fastrtps.repos`](https://github.com/eProsima/Fast-DDS/blob/master/fastrtps.repos) that includes the versions for the related eProsima's maintained dependencies. Other dependencies may be set taking into account the [thirdparty](https://github.com/eProsima/Fast-DDS/tree/master/thirdparty) folder or the version installed by default in the corresponding platform.
